api/src/main/java/net/kyori/adventure/text/format/StyleImpl.java

```diff
@@ -117,56 +105,29 @@ static void decorate(final Builder builder, final TextDecoration[] decorations)
 
   @Override
   public TextDecoration.@NotNull State decoration(final @NotNull TextDecoration decoration) {
-    if (decoration == TextDecoration.BOLD) {
-      return this.bold;
-    } else if (decoration == TextDecoration.ITALIC) {
-      return this.italic;
-    } else if (decoration == TextDecoration.UNDERLINED) {
-      return this.underlined;
-    } else if (decoration == TextDecoration.STRIKETHROUGH) {
-      return this.strikethrough;
-    } else if (decoration == TextDecoration.OBFUSCATED) {
-      return this.obfuscated;
+    final @Nullable TextDecoration.State state = this.decorations.get(decoration);
+    if (state != null) {
+      return state;
     }
     throw new IllegalArgumentException(String.format("unknown decoration '%s'", decoration));
   }
 
   @Override
   public @NotNull Style decoration(final @NotNull TextDecoration decoration, final TextDecoration.@NotNull State state) {
     requireNonNull(state, "state");
```

Comment:
Do we need a non-null check on the decoration too?

Reply:
null cases are handled by `StyleImpl::decoration(TextDecoration)` when it fetches the state